terraform/api_gateway.tf.json.template.py

```diff
@@ -116,6 +116,51 @@ def for_domain(cls, domain):
     'status': '$context.status'
 }
 
+file_download_limit = config.waf_file_download_limit
+if file_download_limit is None:
+    file_download_rate_rule = ()
+else:
+    window_length = 60 * 10
+    limit, concurrency = file_download_limit
+    file_download_rate_rule = [{
+        'name': 'FileDownloadRateLimit',
+        'action': {
+            'block': {
+                'custom_response': {
+                    'response_code': 429,
+                    'response_header': [
+                        {
+                            'name': 'Retry-After',
+                            'value': str(round(window_length / limit * concurrency)),
```

Comment:
I'd be happy for this calculation to be done in a property of`config`. We could also encode the windows in the value of AZUL_FILE_DOWNLOAD_RATE_LIMIT with a note that the current minimum window is 10 minutes.